pkg/network/ebpf/c/protocols/classification/shared-tracer-maps.h

```diff
@@ -35,4 +35,67 @@ __maybe_unused static __always_inline void update_protocol_stack(conn_tuple_t* s
     set_protocol(stack, cur_fragment_protocol);
 }
 
+__maybe_unused static __always_inline void delete_protocol_stack(conn_tuple_t* normalized_tuple, protocol_stack_t *stack, u8 deletion_flag) {
+    if (!normalized_tuple) {
+        return;
+    }
+
+    if (!stack) {
+        stack = bpf_map_lookup_elem(&connection_protocol, normalized_tuple);
+        if (!stack) {
+            return;
+        }
+    }
+
+    if (!(stack->flags&FLAG_USM_ENABLED) || !(stack->flags&FLAG_NPM_ENABLED)) {
+        // If either USM is disabled or NPM is disabled, we cant move rightaway
+        // to the deletion code since there is no chance of race between the two
+        // programs.
+        //
+        // There are two expectected scenarios where just one of the two programs
+        // is enabled:
+        //
+        // 1) When one of the programs is disabled by choice (via configuration);
+        //
+        // 2) During system-probe startup: when system-probe is initializing
+        // there is a short time window where the socket filter programs runs
+        // alone *before* the `tcp_close` probe is activated. In a host with a
+        // network-heavy workload this could easily result in thousands of
+        // leaked entries.
+        goto deletion;
```

Comment:
you stated in the beginning 
> if either USM is disabled or NPM is disabled, we cant move rightaway
to the deletion code since there is no chance of race

but then you're going to "deletion"
did I misunderstood the intention here?

Reply:
argh, it's a typo. I actually meant the opposite: if one of the programs is disabled we _CAN_ move right away to the deletion 🤦